cfpb/consumerfinance.gov
# Add new story page type

Adds a new "Story Page" content type with a sidebar-less centered column and images that bleed into the left and right gutters. It's use case is prose-heavy landing pages designed to tell a story.

To let editors choose when they want either RichTextBlock images or FullWidthText images to bleed into the gutters, a 'bleed' format option has been added to both image fields. Selecting this option for an image that's not on a story page will have no affect (it'll be treated as a standard full width image).

See context at https://github.local/CFPB/el-camino/issues/406

## Notes

- It'd be ideal to only show the "bleed" image format when editing a story page but I don't think that's possible given how the [formats are defined](https://docs.wagtail.io/en/stable/advanced_topics/customisation/page_editing_interface.html#rich-text-image-formats).
- Open to any styling recommendations. I'm using a [content modifier class](https://github.com/cfpb/consumerfinance.gov/compare/story-page?expand=1#diff-388c7690c0d544528764b456f7f7b7b76aa7758049ff55d062338dcf11c32083R22) to scope the CSS. Would a template level class (e.g. `.t-story`) be more appropriate? Or something else altogether? The `.bleed-image` class is vaguely named because it has to apply to both content-image molecules and rich text field images.

## Additions

- Story page type

## Changes

- "Bleed" image format option to both rich text images and full width text images.

## How to test this PR

1. I made a demo page here: https://cfgov-pr-6460.DEMO-PROXY/lorem-ipsum-dolor-sit-amet/
1. Log in with admin/admin to view the editing experience

or

1. Create a Story page.
1. Add a few full width text blocks.
1. Add images and rich text fields with images and they should have a new "bleed" image format.
1. Preview the page and the appropriate images should hang into the left/right gutters.

## Screenshots

| full-width text image selector | rich text image selector | preview |
|--------|--------|---------|
| <img width="711" alt="editor" src="https://user-images.githubusercontent.com/1060248/119527317-35a47080-bd4e-11eb-9e61-da4617769086.png">   | <img width="711" alt="editor" src="https://user-images.githubusercontent.com/1060248/119527334-39d08e00-bd4e-11eb-93d3-37da1ec1d1e7.png">   | <img width="716" alt="preview" src="https://user-images.githubusercontent.com/1060248/119527365-42c15f80-bd4e-11eb-9ab2-3d3e1858e23c.png">    |

## Checklist

  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

### Front-end testing



#### Browser testing

Visually tested in the following supported browsers:



#### Accessibility


#### Other
